src/keras_helper.py

```diff
@@ -62,6 +63,17 @@ def train_model(self, x_train, y_train, epoch=5, batch_size=128, validation_spli
                                                               test_size=validation_split_size)
         self.classifier.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])
 
+
+        # early stopping will auto-stop training process if model stops learning after 2 epochs
+        earlyStopping = EarlyStopping(monitor='val_loss', patience=2, verbose=0, mode='auto')
+
+        # checkpoint for saving best model weights, rather than using the score in the last epoch which isn't necessarily the best score
+        # we can load those best weights for test prediction with: model.load_weights("weights.best.hdf5")
+        
+        filepath="weights.best.hdf5"
+        checkpoint = ModelCheckpoint(filepath, monitor='val_acc', verbose=1, save_best_only=True, mode='max')
```

Comment:
As for the `ModelCheckpoint` I would not write it directly here but rather let the user of the class add it through the `train_callbacks` parameters (which is: moving this definition in the notebook) as:
 - Not everyone wants to use it
 - It does not affect the performances of the model
 - People may want to save it under a different location (in my case I use remote code execution and I need to set the path relative to where the code was uploaded)

What do you think?